In [1]:
from sklearn import model_selection
from sklearn.naive_bayes import MultinomialNB, GaussianNB, BernoulliNB
from sklearn.datasets import load_digits, load_breast_cancer
import matplotlib.pyplot as plt
import numpy as np

In [2]:
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [3]:
data1 = load_digits()
data1.data

array([[  0.,   0.,   5., ...,   0.,   0.,   0.],
       [  0.,   0.,   0., ...,  10.,   0.,   0.],
       [  0.,   0.,   0., ...,  16.,   9.,   0.],
       ..., 
       [  0.,   0.,   1., ...,   6.,   0.,   0.],
       [  0.,   0.,   2., ...,  12.,   0.,   0.],
       [  0.,   0.,  10., ...,  12.,   1.,   0.]])

In [4]:
data2 = load_breast_cancer()
data2.data

array([[  1.79900000e+01,   1.03800000e+01,   1.22800000e+02, ...,
          2.65400000e-01,   4.60100000e-01,   1.18900000e-01],
       [  2.05700000e+01,   1.77700000e+01,   1.32900000e+02, ...,
          1.86000000e-01,   2.75000000e-01,   8.90200000e-02],
       [  1.96900000e+01,   2.12500000e+01,   1.30000000e+02, ...,
          2.43000000e-01,   3.61300000e-01,   8.75800000e-02],
       ..., 
       [  1.66000000e+01,   2.80800000e+01,   1.08300000e+02, ...,
          1.41800000e-01,   2.21800000e-01,   7.82000000e-02],
       [  2.06000000e+01,   2.93300000e+01,   1.40100000e+02, ...,
          2.65000000e-01,   4.08700000e-01,   1.24000000e-01],
       [  7.76000000e+00,   2.45400000e+01,   4.79200000e+01, ...,
          0.00000000e+00,   2.87100000e-01,   7.03900000e-02]])

In [5]:
clf = MultinomialNB()
X,y = load_digits(return_X_y=True)
X1,y1 = load_breast_cancer(return_X_y=True)
print model_selection.cross_val_score(clf, X, y).mean()
print model_selection.cross_val_score(clf, X1, y1).mean()

0.870877148974
0.894579040193


In [6]:
clf = GaussianNB()
X,y = load_digits(return_X_y=True)
X1,y1 = load_breast_cancer(return_X_y=True)
print model_selection.cross_val_score(clf, X, y).mean()
print model_selection.cross_val_score(clf, X1, y1).mean()

0.818600380355
0.936749280609


In [7]:
clf = BernoulliNB()
X,y = load_digits(return_X_y=True)
X1,y1 = load_breast_cancer(return_X_y=True)
print model_selection.cross_val_score(clf, X, y).mean()
print model_selection.cross_val_score(clf, X1, y1).mean()

0.825823650778
0.627420402859


1)Максимальное значение на breast_cancer 0.936749280609

2)Максимальное значение на digits 0.870877148974

3) Верны утверждения (c),(d)